client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -306,6 +307,34 @@ private long getRemainingBytes(SegmentMetadataClientFactory metaFactory, StreamC
         return cuts;
     }
 
+    @Override
+    public CompletableFuture<Map<Stream, StreamCut>> getCurrentStreamCut(ScheduledExecutorService backgroundExecutor) {
+        String internalChkPointId = UUID.randomUUID().toString();
+        log.debug("Fetching the current StreamCut using id {}", internalChkPointId);
+        synchronizer.updateStateUnconditionally(new CreateCheckpoint(internalChkPointId, true));
+        AtomicBoolean fetchStreamCutPending = new AtomicBoolean(true);
+
+        return Futures.loop(fetchStreamCutPending::get, () -> {
```

Comment:
This function shares some code with the other checkpoint function. Perhaps a private method would help

Reply:
done.

Reply:
done.